pkgs/development/compilers/chicken/4/chicken.nix

```
@@ -21,24 +21,32 @@ stdenv.mkDerivation {
     sha256 = "0hvckhi5gfny3mlva6d7y9pmx7cbwvq0r7mk11k3sdiik9hlkmdd";
   };
 
-  setupHook = lib.ifEnable (bootstrap-chicken != null) ./setup-hook.sh;
+  setupHook = lib.optional (bootstrap-chicken != null) ./setup-hook.sh;
 
-  buildFlags = [ "PLATFORM=${platform}" "PREFIX=$(out)" "VARDIR=$(out)/var/lib" ];
-  installFlags = [ "PLATFORM=${platform}" "PREFIX=$(out)" "VARDIR=$(out)/var/lib" ];
+  # -fno-strict-overflow is not a supported argument in clang on darwin
+  hardeningDisable = (lib.optionals stdenv.isDarwin ["strictoverflow"]);
```

```suggestion
  hardeningDisable = lib.optionals stdenv.isDarwin [ "strictoverflow" ];
```
I know, it's just a style preference, but you do add some unneeded parenthesis.